weave/flow/eval.py

```diff
@@ -0,0 +1,187 @@
+import asyncio
+import time
+import inspect
+import traceback
+import typing
+from typing import Any, Callable, Optional, Union
+import numpy as np
+
+import weave
+from weave import op_def
+from weave.flow import Object, Dataset, Model
+from weave.flow.scorer import Scorer, get_scorer_attributes, auto_summarize
+from weave.weaveflow import util
+
+from rich.console import Console
+from rich import print
+
+console = Console()
+
+
+def async_call(
+    func: typing.Union[Callable, op_def.OpDef], *args: Any, **kwargs: Any
+) -> typing.Coroutine:
+    is_async = False
+    if isinstance(func, op_def.OpDef):
+        is_async = inspect.iscoroutinefunction(func.raw_resolve_fn)
+    else:
+        is_async = inspect.iscoroutinefunction(func)
+    if is_async:
+        return func(*args, **kwargs)  # type: ignore
+    else:
```

Comment:
Optional: my personal style preference would be to get rid of the else and deindent the next return.